pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go

```diff
@@ -398,53 +399,65 @@ func (r *ReconcileLinstorSatelliteSet) reconcileAllNodesOnController(ctx context
 
 	linstorClient, err := r.getLinstorClient(ctx, satelliteSet)
 	if err != nil {
-		return []error{err}
+		return err
 	}
 
 	err = r.controllerReachable(ctx, linstorClient)
 	if err != nil {
-		return []error{&reconcileutil.TemporaryError{
+		return &reconcileutil.TemporaryError{
 			Source:       fmt.Errorf("failed to contact controller: %w", err),
 			RequeueAfter: connectionRetrySeconds * time.Second,
-		}}
+		}
 	}
 
 	pods, err := r.getAllNodePods(ctx, satelliteSet)
 	if err != nil {
-		return []error{err}
+		return err
+	}
+
+	k8sNodes := &corev1.NodeList{}
+
+	err = r.client.List(ctx, k8sNodes)
+	if err != nil {
+		return fmt.Errorf("failed to get kubernetes nodes: %w", err)
 	}
 
 	// Every registration routine gets its own channel to send a list of errors
-	outputs := make([]chan error, len(pods))
+	g, gCtx := errgroup.WithContext(ctx)
 
 	for i := range pods {
 		pod := &pods[i]
-		output := make(chan error)
-		outputs[i] = output
+
+		k8sNode := findK8sNode(k8sNodes.Items, pod.Spec.NodeName)
+
+		if k8sNode == nil {
+			logger.WithField("pod", pod.Name).Debug("Node for pod not found, assuming node deleted.")
+
+			continue
+		}
 
 		// Registration can be done in parallel, so we handle per-node work in a separate go-routine
-		go func() {
-			defer close(output)
-			output <- r.reconcilePod(ctx, linstorClient, satelliteSet, pod)
-		}()
+		g.Go(func() error {
+			return r.reconcilePod(gCtx, linstorClient, satelliteSet, pod, k8sNode)
+		})
 	}
 
-	logger.Debug("start collecting per-node reconciliation results")
+	err = g.Wait()
```

Comment:
This only returns the first error, hiding the others. Previously we collected them all. Do we want this?